File: service-common/src/main/java/me/reidj/bridgebuilders/packages/BulkSaveUserPackage.java
package me.reidj.bridgebuilders.packages;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;
import ru.cristalix.core.network.CorePackage;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = false)
@RequiredArgsConstructor
public class BulkSaveUserPackage extends CorePackage {

    // request
    private final List<SaveUserPackage> packages;

    // no response
}

File: service-common/src/main/kotlin/me/reidj/bridgebuilders/protocol/BulkSaveUserPackage.kt
package me.reidj.bridgebuilders.protocol

import ru.cristalix.core.network.CorePackage

/**
 * @project : BridgeBuilders
 * @author : Рейдж
 **/
data class BulkSaveUserPackage(val packages: List<SaveUserPackage>): CorePackage()
